packages/task/src/browser/task-service.ts

```diff
@@ -655,8 +674,13 @@ export class TaskService implements TaskConfigurationClient {
             const terminalId = matchedRunningTaskInfo.terminalId;
             if (terminalId) {
                 const terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
-                if (terminal) {
-                    this.shell.activateWidget(terminal.id); // make the terminal visible and assign focus
+                if (terminal &&
+                    task.presentation && (task.presentation.focus || task.presentation.reveal === RevealKind.Always)
+                ) {
+                    // make the terminal visible and assign focus, if
+                    // 1) task.presentation.focus === true, or
+                    // 2) task.presentation.reveal === 'always'
+                    this.shell.activateWidget(terminal.id);
```

Comment:
reveal means focus? Should not be `shell.revealWidget` used for presentation.reveal without presentation.focus?

Reply:
This is what vscode does 

https://github.com/microsoft/vscode/blob/26076a4de974ead31f97692a0d32f90d735645c0/src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts#L215-L218

When I had ```"reveal": "always", "focus": false``` in the config, the terminal still gets the focus.

if you think we should differentiate "revealWidget" from "activateWidget" i can do that :)

Reply:
> This is what vscode does

They pass down `focus` to `openPanel` and only if `focus` is true then they focus. Otherwise they do nothing but only `setActiveInstance` which only changes the visibility. Do you see something else?

Reply:
Like I said the terminal gets focus even if "focus" is false:

![Peek 2020-01-03 20-15](https://user-images.githubusercontent.com/37082801/71757711-c2067880-2e65-11ea-933c-66b9b7d00975.gif)

Reply:
> The above GIF shows what the terminal widget looks like when I had "reveal": "always", "focus": false

I see on the gif that focus still belongs to the editor, i.e. look at the caret or try typing.

>  However in theia, as a user I don't see the way to make a terminal invisible (unless) closing it.

Don't get why do you need to make it invisible? I thought if `focus` is `true` then use `activateWidget` if not then check `reveal` and if it evaluates to `true` then use `revealWidget`.

Reply:
Thank you for the guidance !

I updated the code.